### DISCUSSION TITLE
Cannot change the name displayed when connecting to a client

### DISCUSSION THREAD

**Discussion by digdug3** [rustdesk/rustdesk#3507] [Q&A] [answered]
### Description

In Teamviewer and AnyDesk it's possible to change the displayname when connecting to a client.
Now Rustdesk just takes the username, would be great if we can change it before connecting and/or changing it in the config (rustdesk.toml) file.

### Impact

When connecting using a computer that's not yours or when your username is acually a nickname, the client will see an unknown username. Allowing to change the user/displayname fixes this.

**Comment by rustdesk** [comment]
A snapshot please, frankly, i did not get your said.

**Reply by My1** [reply]
a screenshot was posted (albeit as a new reply which is why you might not have got any notifs about that (I'll attach it below anyway just in case)

is there a way to change that name?


![](https://user-images.githubusercontent.com/2038575/222902413-a8d1de44-278f-4c91-a0e8-593aabdc658a.png)

**Comment by digdug3** [comment]
I would like to change "john" into something else:
![Screenshot 2023-03-04 134321](https://user-images.githubusercontent.com/2038575/222902413-a8d1de44-278f-4c91-a0e8-593aabdc658a.png)

**Comment by kaypoush** [comment]
my true name just got leaked because of this please make an option to change the name or by default show the ID not the user name

**Comment by trymeouteh** [comment]
Would like this feature as well. A way to change your username and other personal information in the app itself and within the config file itself.

**Comment by rso42** [comment]
If the client I‘m connected to could show my name AND my company‘s name … that would make the still relatively unknown remote control software much more trustworthy for my clients.

**Comment by powellnorma** [comment]
Relevant code is here:
https://github.com/rustdesk/rustdesk/blob/1.2.3-1/src/client.rs#L1738
https://github.com/rustdesk/rustdesk/blob/1.2.3-1/src/common.rs#L744

On linux, in the end `getpwuid_r` is used:
https://github.com/ardaku/whoami/blob/v1.5.0-pre.0/src/os/unix.rs#L207

So a workaround using `bwrap`
```
cp /etc/passwd ~/passwd-nickname
bwrap --dev-bind / / --bind ~/passwd-nickname /etc/passwd /usr/bin/rustdesk
```

**Reply by Kurgan-** [reply]
THIS IS THE REAL ANSWER. It works just fine.

**Comment by sesom42** [comment]
This is an essential feature for professional use. Our company-internal login names may not be displayed externally to our customers.

**Comment by MajliTech** [comment]
I second this.

**Comment by q1ngyang** [comment]
I also hope to implement this feature. 
Currently, displaying username@hostname when connecting with strangers easily leaks private information. 
I hope it can be customized on the client side.

**Comment by kylemccoy** [comment]
I think this is a must-have feature, please implement this...

**Comment by TXSevenXT** [comment]
Hello Team,

Thank a lot for your great work =)

This features seems to be missing (maybe not).
I've seen "Change ID" setting into Security settings but it's greyed :(

What is the condition in order to use it ?

Have a nice day =D

**Reply by My1** [reply]
the id has nothing to do with the name it shows. the ID is the number you give to the other person.

**Comment by JohnnyHenver** [comment]
This is an important feature for professional use cases. It would be best if the administrator could define how remote administrators are shown for clients. The current situation, where OS usernames are used, is not great since connection requests do not look very trustworthy to clients.

**Comment by skythe83** [comment]
It is using my windows 11 username, which was chosen for me, without a chance to set or change.
See for yourself why this cannot be what the customer sees:
![image](https://github.com/rustdesk/rustdesk/assets/32955452/97844230-2056-42fb-90c7-da0e5b253a7f)

**Reply by My1** [reply]
In case of Microsoft accounts it's usually the first 5 of your email address

**Reply by skythe83** [reply]
I know. That doesn't make it changeable though, not without putting everything on its head.

**Comment by Kurgan-** [comment]
This is an issue for me, too.  I'd really like to be able to show a different username.

**Comment by rustdesk** [comment, marked as answer]
https://github.com/rustdesk/rustdesk-server-pro/issues/277

**Comment by kenaucly** [comment]
I think issue was not solved, still cannot change username in not pro version of rustdesk.

**Comment by W-i-n-7** [comment]
how can i change the username or even show just the ID
using system name is stupid

**Reply by Kurgan-** [reply]
You cannot unless you modify and recompile the open source version, or you buy the pro version.

**Reply by W-i-n-7** [reply]
> You cannot unless you modify and recompile the open source version, or you buy the pro version.

i do not understand why rustdesk is missing very basic features like this

**Reply by Kurgan-** [reply]
Because the less features are available in the free version, the more licenses for the pro version are sold.
If you use Linux, this trick works perfectly https://github.com/rustdesk/rustdesk/discussions/3507#discussioncomment-8577826
